runsc/cgroup/cgroup_test.go

```diff
@@ -653,60 +730,88 @@ func TestPids(t *testing.T) {
 
 func TestLoadPaths(t *testing.T) {
 	for _, tc := range []struct {
-		name    string
-		cgroups string
-		want    map[string]string
-		err     string
+		name      string
+		cgroups   string
+		mountinfo string
+		want      map[string]string
+		err       string
 	}{
 		{
-			name:    "abs-path",
-			cgroups: "0:ctr:/path",
-			want:    map[string]string{"ctr": "/path"},
+			name:      "abs-path",
+			cgroups:   "0:ctr:/path",
+			mountinfo: debianMountinfo,
+			want:      map[string]string{"ctr": "/path"},
 		},
 		{
-			name:    "rel-path",
-			cgroups: "0:ctr:rel-path",
-			want:    map[string]string{"ctr": "rel-path"},
+			name:      "rel-path",
+			cgroups:   "0:ctr:rel-path",
+			mountinfo: debianMountinfo,
+			want:      map[string]string{"ctr": "rel-path"},
 		},
 		{
-			name:    "non-controller",
-			cgroups: "0:name=systemd:/path",
-			want:    map[string]string{"systemd": "/path"},
+			name:      "non-controller",
+			cgroups:   "0:name=systemd:/path",
+			mountinfo: debianMountinfo,
+			want:      map[string]string{"systemd": "path"},
 		},
 		{
-			name: "empty",
+			name:      "empty",
+			mountinfo: debianMountinfo,
 		},
 		{
 			name: "multiple",
 			cgroups: "0:ctr0:/path0\n" +
 				"1:ctr1:/path1\n" +
 				"2::/empty\n",
+			mountinfo: debianMountinfo,
 			want: map[string]string{
 				"ctr0": "/path0",
 				"ctr1": "/path1",
 			},
 		},
 		{
-			name:    "missing-field",
-			cgroups: "0:nopath\n",
-			err:     "invalid cgroups file",
+			name:      "missing-field",
+			cgroups:   "0:nopath\n",
+			mountinfo: debianMountinfo,
+			err:       "invalid cgroups file",
 		},
 		{
-			name:    "too-many-fields",
-			cgroups: "0:ctr:/path:extra\n",
-			err:     "invalid cgroups file",
+			name:      "too-many-fields",
+			cgroups:   "0:ctr:/path:extra\n",
+			mountinfo: debianMountinfo,
+			err:       "invalid cgroups file",
 		},
 		{
 			name: "multiple-malformed",
 			cgroups: "0:ctr0:/path0\n" +
 				"1:ctr1:/path1\n" +
 				"2:\n",
-			err: "invalid cgroups file",
+			mountinfo: debianMountinfo,
+			err:       "invalid cgroups file",
+		},
+		{
+			name: "nested-cgroup",
+			cgroups: `9:memory:/docker/136811d8fa1136e2746d10f6443c4c787c3cfbab5273270cc3aeeb3a94b3cc51
+2:cpu,cpuacct:/docker/136811d8fa1136e2746d10f6443c4c787c3cfbab5273270cc3aeeb3a94b3cc51
+1:name=systemd:/docker/136811d8fa1136e2746d10f6443c4c787c3cfbab5273270cc3aeeb3a94b3cc51
+0::/system.slice/containerd.service`,
+			mountinfo: dindMountinfo,
+			// we want relative path to /sys/fs/cgroup inside the nested container.
+			// Subcroup inside the container will be created at /sys/fs/cgroup/cpu
+			// This will be /sys/fs/cgroup/cpu/docker/136811d8fa1136e2746d10f6443c4c787c3cfbab5273270cc3aeeb3a94b3cc51/CGROUP_NAME
+			// outside the container
+			want: map[string]string{
```

Comment:
This is good case to test. Please add another one where the resulting path is a directory name.

Reply:
done.